### ISSUE TITLE
mac10.15,pyinstaller ,how to deal ...registry.txt

### ISSUE THREAD

**Issue by door5719** [librosa/librosa#1289]
In mac10.15, python3.7.10,librosa ==0.8.0
I just use code as：
...
from PyQt5 import QtCore
import librosa
import sys 
...

and command line : pyinstaller -F player.py
completed successfully.
After this process ,i run ./player, I get as follow
,

Traceback (most recent call last):
  File "player.py", line 5, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 531, in exec_module
  File "librosa/__init__.py", line 211, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 531, in exec_module
  File "librosa/core/__init__.py", line 5, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 531, in exec_module
  File "librosa/core/convert.py", line 7, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 531, in exec_module
  File "librosa/core/notation.py", line 8, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 531, in exec_module
  File "librosa/util/__init__.py", line 84, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 531, in exec_module
  File "librosa/util/files.py", line 34, in <module>
  File "pooch/core.py", line 600, in load_registry
FileNotFoundError: [Errno 2] No such file or directory: '/var/folders/7s/qrf9bqgj7k988g8t0g7m_vrr0000gn/T/_MEIa7KVri/librosa/util/example_data/registry.txt'

how to deal with this?

**Comment by lostanlen** [comment]
Related: https://stackoverflow.com/questions/65833533/python-registry-txt-not-found-when-librosa-load-used-in-pyinstaller-exe

**Comment by bmcfee** [comment]
This indeed looks like a problem with pyinstaller not correctly detecting the data files specified in our packaging config https://github.com/librosa/librosa/blob/e11a7cc360c430001e851855b97dcf3836098a78/setup.py#L21

I'm inclined to mark this up as a bug in pyinstaller, though it's possible that there may be a way to change the package config to work around it.  It wouldn't take effect until the next release, but it's something to keep in mind.  I'll leave this open for now, and revisit when we're doing release prep.

For now, my best advice is to not use pyinstaller.

**Comment by bmcfee** [comment]
This is pretty weird.  I double-checked the archive file on pypi, and it does indeed include all the correct files.

Then I checked our 0.8.1rc1 archive, and they're included in the wheel but not the sdist.  This must have happened when I switched over to pyproject.toml for packaging, but it's a known and expected behavior (apparently).  Why this didn't hit us before, I don't entirely know, but I've amended the manifest over in #1339 to ensure that package data is always included.

This fix, combined with the wheel for 0.8.1rc1, ought to fix the pyinstaller issue.  It'd be great if @door5719 could confirm, but it might require hacking a spec file to get pyinstaller to use a pre-release package.

**Comment by bmcfee** [comment]
@door5719 if you have a moment, can you check whether this issue still occurs with the 0.8.1 release?  We've redone the packaging configuration, so it should resolve the issue, but I'd like to confirm before closing this out.

**Comment by godisloveforme** [comment]
@bmcfee I can still reproduce this with 0.8.1 release.

**Comment by bmcfee** [comment]
> @bmcfee I can still reproduce this with 0.8.1 release.

Bummer.  Can you provide a bit more information here (versions, log file, etc)?

Looking into the [pyinstaller docs](https://pyinstaller.readthedocs.io/en/stable/usage.html), it seems like adding `--collect-data librosa` might do the trick.

**Comment by bmcfee** [comment]
It looks to me like `--collect-data librosa` does the trick, and this is all basically expected behavior from pydata.

I haven't gotten pyinstaller to produce a binary that actually runs (shared object problems, for some reason), but I've inspected the spec files and manifests with and without the extra option.

Here's a basic test script:
```python
#!/usr/bin/env python
import librosa
librosa.show_versions()
```

When running `pyinstaller test.py` and then checking `build/test/`, I see the following:
```
 →  grep registry.txt *
 1  🐍 py39  ~/pyinstalltest/build-nodata/test 
```
If I run `pyinstaller --collect-data librosa test.py` and then check again, I see the data showing as expected:
```
 →  grep registry.txt *
COLLECT-00.toc:  ('librosa/util/example_data/registry.txt',
COLLECT-00.toc:   '/home/bmcfee/git/librosa/librosa/util/example_data/registry.txt',
 🐍 py39  ~/pyinstalltest/build/test 
```

So to summarize, it seems like we're doing everything correctly, and pyinstaller just doesn't include data files unless explicitly asked.  If there are no further objections, I'll close this issue out.

**Comment by drfeinberg** [comment]
Hi.  I'm having the same issue on windows. That solution did not work, but this did:
https://stackoverflow.com/questions/63686431/filenotfounderror-errno-2-no-such-file-or-directory-pyinstaller-searching-fo
replacing `librosa` for `cmudict`
